shared/fs/utils/dispatch-mappers.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import {type Dispatch} from '../../util/container'
+import * as FsGen from '../../actions/fs-gen'
+import * as Types from '../../constants/types/fs'
+import {navigateAppend, navigateUp} from '../../actions/route-tree'
+import {isMobile} from '../../constants/platform'
+
+export const mapDispatchToShowInFileUI = (dispatch: Dispatch) => (path: Types.Path) =>
+  dispatch(FsGen.createOpenInFileUI({path: Types.pathToString(path)}))
+
+export const mapDispatchToShare = (dispatch: Dispatch) => (path: Types.Path) =>
```

Comment:
another option is to just make an action which just does this as a side effect

Reply:
Hmm this is to dispatch a `navigateAppend` to the route tree. You mean we'd use Saga to call dispatch?